### PR TITLE
Validate redis-server/redis-cli install

### PR DESCRIPTION
**Pull request by billschereriii**
Add validation of redis-server and redis-cli installation, based on how we'll expect to find them in config, to builder class

### CODE REVIEW

**Comment by billschereriii** [comment]
Not happy about wholesale cloning of `expand_exe_path()` -- can someone suggest a better option?

File: smartsim/_core/_install/builder.py [hunk 1]
```diff
@@ -15,6 +16,25 @@
 #   - check cmake version and use system if possible to avoid conflicts
 
 
+def expand_exe_path(exe):
```
Comment (ashao) [inline smartsim/_core/_install/builder.py:19]: While duplicating code is no fun, for the case of things that are used for `setup.py` and all that, it's better to err on the side of keeping it clean. As we discussed briefly, the only thing that I can think of is to import the function in the utils from here and make it available. I say keep it as is, unless @al-rigazzi has any objections.

**Review by ashao — APPROVED** [review]
I can confirm that this should not impact the site-installs (so long as the user has followed the instructions and imported the modules beforehand). Thanks for getting this in @billschereriii

**Review by al-rigazzi — APPROVED** [review]
LGTM, thanks @billschereriii, this will make it easier to spot broken installs!

**Pull request merged**